pandas/core/resample.py

```diff
@@ -1346,18 +1351,35 @@ def __init__(
         self.convention = convention or "E"
         self.convention = self.convention.lower()
 
-        if isinstance(loffset, str):
-            loffset = to_offset(loffset)
-        self.loffset = loffset
-
         self.how = how
         self.fill_method = fill_method
         self.limit = limit
-        self.base = base
+
+        if origin in {"epoch", "start_day", "start"}:
+            self.origin = origin
+        else:
+            self.origin = Timestamp(origin)
```

Comment:
you might want to change this sligthly
```
if origin in ....:
    self.origin = origin
else:
    try:
       self.origin = Timestamp(origin)
    except Exception as e:
        raise ValueError(....nice message here) from e
```

Reply:
I have added tests as long with a nice ValueError message for `origin` and `offset`

Here:
https://github.com/pandas-dev/pandas/blob/f2458dfaf5bdc22e22f58f7b720fa206b3cd88ef/pandas/tests/resample/test_datetime_index.py#L766

And here:
https://github.com/pandas-dev/pandas/blob/f2458dfaf5bdc22e22f58f7b720fa206b3cd88ef/pandas/tests/resample/test_datetime_index.py#L780